src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,17 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer<E extends Machine> implements MachineProducer<Excavator> {
```

Comment:
What is the purpose to generify ExcavatorProducer with E extends Machine? You don't use "E" in code below.

Reply:
It was a technical mistake, I was playing with different wildcards and didn't remove this one :(